relayer/chainproxy/portalLogs.go

```diff
@@ -96,9 +103,38 @@ func (pl *PortalLogs) LogStartTransaction(name string) {
 	}
 }
 
-func (pl *PortalLogs) AddMetric(data *metrics.RelayMetrics, isNotSuccessful bool) {
-	if pl.StoreMetricData {
-		data.Success = !isNotSuccessful
-		pl.MetricService.SendData(*data)
+func (pl *PortalLogs) AddMetric(data *metrics.RelayMetrics, isSuccessful bool) {
+	if pl.storeMetricData {
+		data.Success = isSuccessful
+		pl.metricService.SendData(*data)
 	}
 }
+
+func (pl *PortalLogs) ShouldCountMetric(c *fiber.Ctx) bool {
+	refererHeaderValue := c.Get(refererHeaderKey, "")
+	if refererHeaderValue == "" {
+		refererHeaderValue = c.Get(strings.ToLower(refererHeaderKey), "")
+	}
+	if len(pl.toExcludeMetricsReferrers) > 0 && refererHeaderKey != "" {
```

Comment:
this condition should be in one function and used by the 3 -"shouldCount" functions, so no code duplication and if we add/remove conditions we will do it on one place only